dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerAPIImpl.java

```diff
@@ -353,18 +347,62 @@ public void saveContainerStructures(final List<ContainerStructure> containerStru
     @Override
     public List<ContentType> getContentTypesInContainer(Container container) throws DotStateException, DotDataException, DotSecurityException  {
 
-        List<ContainerStructure> csList =  getContainerStructures(container);
-        List<ContentType> cTypeList = new ArrayList<ContentType>();
+		try {
+			List<ContainerStructure>  csList = getContainerStructures(container);
 
-        for (ContainerStructure cs : csList) {
-            ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(cs.getStructureId());
-            if(type==null) continue;
-            cTypeList.add(type);
-        }
-        
-        return cTypeList.stream().sorted(Comparator.comparing(ContentType::name)).collect(Collectors.toList());
+			List<ContentType> cTypeList = new ArrayList<>();
+
+			for (ContainerStructure cs : csList) {
+				try {
+					ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(cs.getStructureId());
+					if(type == null) continue;
+					cTypeList.add(type);
+				} catch (DotSecurityException e) {
+					continue;
+				}
+			}
+
+			return cTypeList.stream().sorted(Comparator.comparing(ContentType::name)).collect(Collectors.toList());
+
+		} catch (DotSecurityException e) {
+			return Collections.EMPTY_LIST;

```

Comment:
can you log the exception?

Reply:
this exception really never happend, this exception is suppose to be throw by  'getContainerStructures' method according its signature:

https://github.com/dotCMS/core/blob/55da87fb8d5803f97bc0d92ffb90fd12ec83baae/dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerAPIImpl.java#L325

but this method really never throw that exception

https://github.com/dotCMS/core/blob/55da87fb8d5803f97bc0d92ffb90fd12ec83baae/dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerAPIImpl.java#L322

Reply:
Also the method return all the content types that can be add by user into a specific container, so if the user can get the container Object then the method return a empty list